src/content/docs/en/reference/integrations-reference.mdx

```diff
@@ -452,10 +462,143 @@ const integration = {
 
 **Previous hook:** [`astro:config:setup`](#astroconfigsetup)
 
+**Next hook:** [`astro:config:done`](#astroconfigdone)
+
+**When:** In `astro dev`, it also runs if a file based route changes (added/removed/updated).
+
+**Why:** To access routes and their metadata
+
+```js
+'astro:routes:resolved'?: (options: {
+  routes: IntegrationResolvedRoute[];
+  logger: AstroIntegrationLogger;
+}) => void | Promise<void>;
+```
+
+#### `routes` option
+
+**Type:** [`IntegrationResolvedRoute[]`](#integrationresolvedroute-type-reference)
+
+A list of all routes alongside their associated metadata.
+
+You can reference the full `IntegrationResolvedRoute` type below, but the most common properties are:
+
+- `entrypoint` - the input file path relative to the project root
+- `pathname` - the output file URL (undefined for routes using `[dynamic]` and `[...spread]` params)
+
+##### `IntegrationResolvedRoute` type reference
+
+```ts
+interface IntegrationResolvedRoute {
+	/**
+	 * The current **pattern** of the route. For example:
+	 * - `src/pages/index.astro` has a pattern of `/`
+	 * - `src/pages/blog/[...slug].astro` has a pattern of `/blog/[...slug]`
+	 * - `src/pages/site/[blog]/[...slug].astro` has a pattern of `/site/[blog]/[...slug]`
+	 */
+	pattern: RouteData['route'];
+
+	/**
+	 *
+	 * regex used for matching an input URL against a requested route
+	 * ex. "[fruit]/about.astro" will generate the pattern: /^\/([^/]+?)\/about\/?$/
+	 * where pattern.test("banana/about") is "true"
+	 *
+	 * ## Example
+	 *
+	 * ```js
+	 * if (route.pattern.test('/blog')) {
+	 *  // do something
+	 * }
+	 * ```
+	 */
+	patternRegex: RouteData['pattern'];
+
+	/**
+	 *  Source component URL
+	 */
+	entrypoint: RouteData['component'];
+
+	/**
+	 * Whether the route is prerendered or not
+	 */
+	isPrerendered: RouteData['prerender'];
+
+	/**
+	 * The {@link IntegrationResolvedRoute} to redirect to. It's present when `IntegrationResolvedRoute.type` is `redirect`.
+	 */
+	redirectRoute?: IntegrationResolvedRoute;
+
+  /**
+	 * @param {any} data The optional parameters of the route
+	 *
+	 * @description
+	 * A function that accepts a list of params, interpolates them with the route pattern, and returns the path name of the route.
+	 *
+	 * ## Example
+	 *
+	 * For a route such as `/blog/[...id].astro`, the `generate` function would return something like this:
+	 *
+	 * ```js
+	 * console.log(generate({ id: 'presentation' })) // will log `/blog/presentation`
+	 * ```
+	 */
+	generate: (data?: any) => string;
+
+  /**
+	 * Dynamic and spread route params
+	 * ex. "/pages/[lang]/[...slug].astro" will output the params ['lang', '...slug']
+	 */
+	params: string[];
+
+  /**
+	 * Output URL pathname where this route will be served
+	 * note: will be undefined for [dynamic] and [...spread] routes
+	 */
+	pathname?: string;
+
+  /**
+	 * Similar to the "params" field, but with more associated metadata. For example, for `/site/[blog]/[...slug].astro`, the segments are:
```

Comment:
```suggestion
	 * An array of metadata for each segment (separated by `/`) of the route pattern. Includes the `content` of the segment, as well as whether or not the path segment is dynamic and spread. For example, for `/site/[blog]/[...slug].astro`, the segments are:
```
OK, you might need to fix this up but this seemed more related to `pattern` in my mind than params, since params don't care about the static segments.